### ISSUE TITLE
Issue with k8s.io/docs/tasks/debug-application-cluster/debug-service/

### ISSUE THREAD

**Issue by cwingrav** [kubernetes/website#7281]
**This is a...** 
- [x] Feature Request
- [ ] Bug Report

**Problem:**
TLDR: New Case to Handle in Documentation: localhost connection worked; svc and endpoint didn't; problem was localhost/0.0.0.0 differences in ExpressJS

I was debugging a service which was not connecting to my pod. I was running a listening nodejs app in one bash shell in my service and trying to connect in another terminal on that pod, via a curl call. I went through the instructions on this page and my service had endpoints and I could call 'curl localhost:50105/health' just fine. However, I could not call the service's endpoint url, 'curl msdev/health' or even the endpoint of that service, 'curl 10.2.87.67:50105/health'. Both resulted in 'connection refused'. I verified with ifconfig that 10.2.87.67 was the pod's IP.

I restarted the listening app multiple times. Nothing changed.

The container base I am using is the ubuntu container from Dockerhub. I at first thought some type of firewall but iptables on the pod showed nothing.

I eventually found that my app was listening on 'localhost' and not '0.0.0.0' (via ExpressJS). This is a common nodeJS package.

**Proposed Solution:**
I suggest a line in the documentation discussing differences in running an app locally versus in Kubernetes. Something that suggests checking your app's listener code for issues in networked operation vs on your own local machine. I know this can easily be considered out of scope of this documentation, but a simple mention of somewhere else to look into would have saved me a lot of time.

I make lots of silly mistakes, but I can't believe I'm the only one... Cheers!

**Page to Update:**

https://kubernetes.io/docs/tasks/debug-application-cluster/debug-service/

Kubernetes Version: 1.8.4
